mindmeld/components/_elasticsearch_helpers.py

```diff
@@ -60,6 +62,26 @@ def create_es_client(es_host=None, es_user=None, es_pass=None):
         raise KnowledgeBaseError
 
 
+def is_es_version_7(es_client):
+    major_version = int(es_client.info()["version"]["number"].split(".")[0])
+    if major_version < 5:
+        logger.warning(
+            "Major version of ElasticSearch %d is not officially supported.",
+            major_version,
+        )
+    if major_version >= 7:
+        return True
+    return False
+
+
+def resolve_es_config_for_version(config, es_client):
+    if not is_es_version_7(es_client):
+        if DOC_TYPE not in config.get("mappings", {}):
```

Comment:
Could you add a comment on why we are doing this logic specially for ES7?

Reply:
Sure added. It is because of the breaking change in ES7, and because we don't want users to have to re-index existing indices that were created with older versions of MindMeld: https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html#removal-of-types